include/podio/utilities/TypeHelpers.h

```diff
@@ -244,8 +247,63 @@ class CollectionBase;
 
 /// Concept for checking whether a passed type T inherits from podio::CollectionBase
 template <typename T>
-concept CollectionType = std::is_base_of_v<CollectionBase, T>;
-
+concept CollectionType = requires(T t, const T ct) {
+  // general concepts
+  requires !std::is_abstract_v<T>;
+  requires std::derived_from<T, CollectionBase>;
+  requires std::default_initializable<T>;
+  requires std::destructible<T>;
+  requires std::movable<T>;
+  requires !std::copyable<T>;
+  requires std::ranges::random_access_range<T>;
+  // typeName's
+  { T::typeName } -> std::convertible_to<std::string_view>;
+  { std::bool_constant<(T::typeName, true)>() } -> std::same_as<std::true_type>; // ~is annotated with constexpr
+  { T::valueTypeName } -> std::convertible_to<std::string_view>;
+  { std::bool_constant<(T::valueTypeName, true)>() } -> std::same_as<std::true_type>; // ~is annotated with constexpr
+  { T::dataTypeName } -> std::convertible_to<std::string_view>;
+  { std::bool_constant<(T::dataTypeName, true)>() } -> std::same_as<std::true_type>; // ~is annotated with constexpr
+  // typedefs
+  typename T::value_type;
+  typename T::mutable_type;
+  requires std::convertible_to<typename T::mutable_type, typename T::value_type>;
+  typename T::difference_type;
+  requires std::signed_integral<typename T::difference_type>;
+  typename T::size_type;
+  requires std::unsigned_integral<typename T::size_type>;
+  typename T::const_iterator;
+  requires std::random_access_iterator<typename T::const_iterator>;
+  typename T::iterator;
+  requires std::random_access_iterator<typename T::iterator>;
```

Comment:
At least this one is already implied by `std::ranges::random_access_range`. Are none of the others already implied? If so they could be removed.